main/util/src/mill/util/SystemStreams.scala

```diff
@@ -2,4 +2,9 @@ package mill.util
 
 import java.io.{InputStream, PrintStream}
 
-class SystemStreams(val out: PrintStream, val err: PrintStream, val in: InputStream)
+class SystemStreams(
+    val out: PrintStream,
+    val err: PrintStream,
+    val in: InputStream,
+    val bspLog: Option[PrintStream] = None
```

Comment:
I'm not sure this is the right place to put this, since it is really only used by the BSP logic and not the rest of the code in the repo. Furthermore, it kind of breaks the encapsulation here, which is that `SystemStreams` is meant to be the `stdout`/`stdin`/`stderr` streams that are provided by the operating system and not include any application-specific values

Reply:
> I'm not sure this is the right place to put this, since it is really only used by the BSP logic and not the rest of the code in the repo. 

Yeah, makes sense. Let me find a better solution.

> Furthermore, it kind of breaks the encapsulation here, which is that `SystemStreams` is meant to be the `stdout`/`stdin`/`stderr` streams that are provided by the operating system and not include any application-specific values

Here I disagree. Initially, `SystemStreams` was created to hold the OS provided streams, but after your refactoring, it's rather a class to pass the three typical streams around. The point in the BSP mode is, that we very early redirect and run the rest of Mill with these customizations. If we want `SystemStreams` to hold the original OS values, we can't use this class further down to pass customized streams around.

Reply:
Oh I don't mean that it's only meant to pass the oringal OS streams around, more that it's meant to pass some abstraction that looks like them and expose the same interface. So they can get substituted or redirected, but they're still meant to have 3 streams. So any code that receives it can use those three streams as if it's the OS-provided ones, even if they got replaced somewhere along the way

Reply:
Understood. Let we work out some alternative for `bspLog`.